shopinvader_api_payment_cart/models/sale_order.py

```diff
@@ -12,8 +12,18 @@ def _action_confirm_cart_from_tx(self, tx):
         """
         Create a callback method that will be called when the related
         payment transaction will be confirmed.
-        This method confirms the cart if transaction succeeded.
+        This method confirms the cart (into a quotation) if transaction succeeded.
+        For custom payments, the transaction will never be done automatically
+        but goes into "pending" state as soon as a wire transfer request was made
+        (as in standard Odoo flow).
+        What's more, if configured so on the payment provider, auto-confirm
+        the quotation.
         """
-        if tx.state == "done":
-            return self.action_confirm_cart()
+        if tx.state == "done" or (
+            tx.provider_id.code == "custom" and tx.state == "pending"
+        ):
+            res = self.action_confirm_cart()
+            if tx.provider_id.shopinvader_auto_confirm_linked_so:
+                self.action_confirm()
+            return res
```

Comment:
Not sure what to do here. This module should have knowledge about specific payment modes.

Reply:
This boolean `shopinvader_auto_confirm_linked_so` allows to configure the providers as the customer wants.
Some of them want the picking to be created, prepared and delivered as soon as the wire transfer payment transaction is pending, but some others want to receive the money before confirming the SO and creating the picking.